src/libraries/System.Private.Xml/src/System/Xml/Core/XmlTextReaderImpl.cs

```diff
@@ -3744,7 +3744,7 @@ private bool ParseXmlDeclaration(bool isTextDecl)
                 switch (_ps.chars[_ps.charPos])
                 {
                     case 'v':
-                        if (XmlConvert.StrEqual(_ps.chars, _ps.charPos, nameEndPos - _ps.charPos, "version") && xmlDeclState == 0)
+                        if (nameEndPos - _ps.charPos == "version".Length && _ps.StartsWith("version") && xmlDeclState == 0)
```

Comment:
I know this was done following recommendations from https://github.com/dotnet/runtime/pull/74955#discussion_r961060685, but this seems to be introducing a lot of duplication. Surely a helper method couldn't hurt here? cc @stephentoub

Reply:
What duplication?

Reply:
Duplication of string literals.

Reply:
This case can be:
```C#
if (_ps.chars.AsSpan()[_ps.charPos..nameEndPos].SequenceEqual("version"))
```
or something like that. Basically, use SequenceEqual when it makes sense and StartsWith when it makes sense.  If cases can be combined into a switch, even better.  The thing to be careful to avoid is doing StartsWith/SequenceEquals that would go beyond the populated portion of the array.